crates/aptos/src/common/types.rs

```diff
@@ -1310,6 +1311,13 @@ impl TransactionOptions {
             gas_unit_price
         };
 
+        // Get sequence number for account
+        let (account, state) = get_account_with_state(&client, sender_address).await?;
+        let sequence_number = account.sequence_number;
+        let expiration_time = state.timestamp_usecs + self.gas_options.expiration_secs * US_IN_SECS;
```

Comment:
I don't think this is the right way to derive an expiration time... this could be several seconds old depending on how quickly the system is processing proposals to finalization.

Reply:
So, there's a couple problems here:
1. How slow is your computer to process it
2. How far off is your clock from the chain
3. You still need to sign, and send the request anyways, which will take some time in itself.

The approach I took here was to use the onchain state, but sure we can error from drift to the local clock which is what most applications do anyways.  I also increased the default timeout anyways, which should decrease any issues associated.